AK/StdLibExtraDetails.h

```diff
@@ -443,8 +462,62 @@ auto declval() -> T;
 
 template<typename T, typename... Args>
 inline constexpr bool IsCallableWithArguments = requires(T t) { t(declval<Args>()...); };
+
+template<typename To>
+To __test_implicit_conversion(To);
+
+template<typename From, typename To>
+inline constexpr bool IsConvertible = requires { __test_implicit_conversion<To>(declval<From>()); };
+
+template<typename T, typename... Args>
+inline constexpr bool IsConstructible = requires { ::new T(declval<Args>()...); };
+
+template<typename T, typename... Args>
+inline constexpr bool IsTriviallyConstructible = __is_trivially_constructible(T, Args...);
+
+template<typename T, typename U>
+inline constexpr bool IsAssignable = requires { declval<T>() == declval<U>(); };
```

Comment:
That doesn't look like `IsAssignable` :)

Reply:
I took this from the C++ standard, which says:

```cpp
template<class T, class U>
struct is_­assignable;
```
> The expression declval<T>() = declval<U>() is well-formed when treated as an unevaluated operand.
> Access checking is performed as if in a context unrelated to T and U.
> Only the validity of the immediate context of the assignment expression is considered.

Reply:
I meant `declval<T>() == declval<U>();` lol

Reply:
yes, but your code says `==`, not `=` :sweat_smile:

Reply:
oh no